File: Backend/Travle.Model/Responses/TagRef.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// Lightweight tag reference (id + name) embedded in richer responses such as
    /// <see cref="DestinationResponse.Tags"/>. The id lets an edit form preselect the tag chips; the
    /// name is shown on screen (never a raw id). The full <see cref="TagResponse"/> with timestamps is
    /// used only by the tag reference-CRUD endpoints.
    /// </summary>
    public class TagRef
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
    }
}

File: UI/travle_core/lib/src/models/tag_ref.dart
import 'package:json_annotation/json_annotation.dart';

part 'tag_ref.g.dart';

/// A lightweight tag reference (id + name) as embedded in a destination (mirrors
/// the backend `TagRef`). The id lets a form preselect the tag chips; the name is
/// what's shown.
@JsonSerializable()
class TagRef {
  TagRef({required this.id, required this.name});

  final int id;
  final String name;

  factory TagRef.fromJson(Map<String, dynamic> json) => _$TagRefFromJson(json);

  Map<String, dynamic> toJson() => _$TagRefToJson(this);
}
